recipes/cmake/3.x.x/conanfile.py

```diff
@@ -37,16 +38,13 @@ def requirements(self):
         if self.options.with_openssl:
             self.requires("openssl/1.1.1q")
 
-    def validate(self):
-        if self.settings.os == "Macos" and self.settings.arch == "x86":
-            raise ConanInvalidConfiguration("CMake does not support x86 for macOS")
-
+    def validate_build(self):
```

Comment:
So we are still trying to iron out the details of how this will work between 1.x and 2.0 (right now this breaks things for 1.x) and thats not the step we are makign (just yet)

Can you please move this to `validate` with the new beta 5 out this should work with `self.settings` and not break 1.x 

CMake is important so i'd rather be safe then sorry on this one

Reply:
Hi @prince-chrismc,
Sorry for the delayed reply. I've gone ahead and eliminated the use of validate_build() in favor of validate(). Indeed, it does seem to work okay now with Conan 2.0 beta 5 when using validate(). That said, I do believe that validate_build() is the more appropriate vehicle for most of the checks being done here as they check the version of the C++ compiler being used to build CMake and (as far as I know) have nothing to do with the use of it as a build tool since it can be used  with both C and C++ compilers (amongst others). Thanks

Reply:
Hi @prince-chrismc,

I needed to restore the use of validate_build() to get these changes to work correctly with Conan 1.x (see below). Thanks.